### PR TITLE
Added spawn ped feature.

### PR DESCRIPTION
**Pull request by aa15032261**
image

### CODE REVIEW

File: BigBaseV2/src/services/gui/gui_service.hpp [hunk 1]
```diff
@@ -59,11 +67,14 @@ namespace big
 					{ tabs::HANDLING_SEARCH, {"Search Handling", view::handling_search } },
 				}}},
 				{ tabs::LSC, { "LS Customs", view::lsc }},
-				{ tabs::SPAWN, { "Spawn", view::spawn }},
+				{ tabs::SPAWN_VEHICLE, { "Spawn Vehicle", view::spawn_vehicle }},
 				{ tabs::PV, { "Personal Vehicle", view::pv }},
-				{ tabs::VEHICLE_FUN, { "Fun Features", view::vehicle_fun }},
 				{ tabs::PERSIST_CAR, { "Persist Car", view::persist_car }},
 			}}},
+			{ tabs::FUN, { "Fun Features", nullptr, {
```
Comment (maybegreat48) [inline BigBaseV2/src/services/gui/gui_service.hpp]: I think all the vehicle stuff should stay in Vehicle and the ped spawner can go in a new category called "World" or something like that

Comment (Yimura) [reply BigBaseV2/src/services/gui/gui_service.hpp]: Yes that does sound better, we can probably move a bunch of things to the world category.

Comment (aa15032261) [reply BigBaseV2/src/services/gui/gui_service.hpp]: Yea, that sounds great. I will change it real quick.

Comment (aa15032261) [reply BigBaseV2/src/services/gui/gui_service.hpp]: This is the updated layout:

image

File: BigBaseV2/src/services/gui/gui_service.hpp [hunk 2]
```diff
@@ -91,7 +102,7 @@ namespace big
 		void set_nav_size(int);
 		void increment_nav_size();
 		void reset_nav_size();
-		std::unordered_map<tabs, navigation_struct> get_navigation();
+		std::map<tabs, navigation_struct> get_navigation();
```
Comment (maybegreat48) [inline BigBaseV2/src/services/gui/gui_service.hpp]: Can't we just return a reference? This is being called every tick

Comment (xiaoxiao921) [reply BigBaseV2/src/services/gui/gui_service.hpp]: And why the change to `map` ?

Comment (maybegreat48) [reply BigBaseV2/src/services/gui/gui_service.hpp]: Probably because iterating an unordered_map returns keys in undefined order

Comment (xiaoxiao921) [reply BigBaseV2/src/services/gui/gui_service.hpp]: Yes I know that but I simply wanted the reasoning behind the change.
Like what bug that change exactly fixed.

Comment (aa15032261) [reply BigBaseV2/src/services/gui/gui_service.hpp]: > Can't we just return a reference? This is being called every tick

Yes, I'll update that as well. I only changed unordered_map to map and didn't pay attention that it was return by value.

Comment (aa15032261) [reply BigBaseV2/src/services/gui/gui_service.hpp]: > Yes I know that but I simply wanted the reasoning behind the change. Like what bug that change exactly fixed.

Map allows us to control the order of the items in the menu, so the order stays the same even if we updated it.

Comment (aa15032261) [reply BigBaseV2/src/services/gui/gui_service.hpp]: get_selected_tab and get_navigation are now return by reference.

**Review by Yimura — DISMISSED** [review]

**Comment by howdoiusekeyboard** [comment]
Add an option to spawn without weapons?

**Comment by aa15032261** [comment]
> Add an option to spawn without weapons?

Thanks for the suggestion! It’s been added in the latest commit.

**Review by Yimura — DISMISSED** [review]

**Review by Yimura — APPROVED** [review]

**Pull request merged**